flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/JobResourceRequirements.java

```diff
@@ -0,0 +1,204 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.runtime.jobmaster;
+
+import org.apache.flink.runtime.jobgraph.JobGraph;
+import org.apache.flink.runtime.jobgraph.JobVertexID;
+import org.apache.flink.util.InstantiationUtil;
+
+import java.io.IOException;
+import java.io.Serializable;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.Optional;
+import java.util.Set;
+
+/** Information about the parallelism of job vertices. */
+public class JobResourceRequirements implements Serializable {
+
+    private static final long serialVersionUID = 1L;
+
+    private static final String JOB_RESOURCE_REQUIREMENTS_KEY = "$job-resource-requirements";
```

Comment:
nit: looks like other keys don't have `$` sign in front

Reply:
We do have `$internal.pipeline.job-id` for example [1] (I'm not aware of other examples 😂 ); an intention was to mark this as internal and not supposed to be exposed to the user. I think changing this to:

```suggestion
    private static final String JOB_RESOURCE_REQUIREMENTS_KEY = "$internal.job-resource-requirements";
```

might be more explicit 🤔 I'll also add a comment with an explanation.

WDYT?

[1] org.apache.flink.configuration.PipelineOptionsInternal](https://github.com/apache/flink/blob/master/flink-core/src/main/java/org/apache/flink/configuration/PipelineOptionsInternal.java)

Reply:
`    private static final String JOB_RESOURCE_REQUIREMENTS_KEY = "$internal.job-resource-requirements";
`
I like it :+1: